Client/Frontend/Menu/MenuConfiguration.swift

```diff
@@ -4,146 +4,171 @@
 
 import Foundation
 
-enum MenuLocation {
-    case Browser
-    case HomePanels
-    case TabTray
+protocol MenuConfiguration {
+
+    var menuItems: [MenuItem] { get }
+    var menuToolbarItems: [MenuToolbarItem]? { get }
+    var numberOfItemsInRow: Int { get }
+
+    init(appState: AppState)
+    func toolbarColour() -> UIColor
+    func toolbarTintColor() -> UIColor
+    func menuBackgroundColor() -> UIColor
+    func menuTintColor() -> UIColor
+    func menuFont() -> UIFont
+    func menuIcon() -> UIImage?
+    func shadowColor() -> UIColor
 }
 
-struct MenuConfiguration {
+struct FirefoxMenuConfiguration: MenuConfiguration {
```

Comment:
Historically, we've avoided using product names (Firefox) directly in code. We already have the `App` prefix (`AppConstants` and now `AppState`), so perhaps `AppMenuItem`/`AppMenuConfiguration` would work here, too?